File: Python/202. Happy Number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        dict_ = {}
        while n != 1:
            dict_[n] = 1
            n = list(str(n))
            temp = 0
            for i in n:
                temp += pow(int(i), 2)
            n = int(temp)
            if dict_.get(n, 0) == 1: return False
        return True

File: C++/202. Happy Number.cpp
class Solution {
public:
    bool isHappy(int n) {
        int limit = 100, m = 0;

        while (n != 1 && limit > 0) {
            m = 0;
            
            while (n) {
                m += (n % 10) * (n % 10);
                n /= 10;
            }

            n = m;
            cout << m << endl;

            limit--;
        }

        return limit > 0;
    }
};
